### ISSUE TITLE
Latex not getting rendered properly

### ISSUE THREAD

**Issue by jukofyork** [ollama-webui/ollama-webui#410]
Some Latex seems to get rendered but most doesn't seem to work, eg:

```
\begin{aligned} \text{Predict:} & x_{k} = f(x_{k-1}) + w_k \\ \text{Update:} & y_{k} = h(x_{k}) + v_k \ \end{aligned}
```

or any use of `$...$` for expressions.

**Comment by tjbck** [comment]
Hi, Thanks for creating this issue! The issue stems from LLMs not generating LaTeX in a correct format. You can observe the same problem with ChatGPT like such:

<img width="696" alt="image" src="https://github.com/ollama-webui/ollama-webui/assets/25473318/63bb2112-25ce-4c21-b265-f69e2dcd98dc">

<img width="432" alt="image" src="https://github.com/ollama-webui/ollama-webui/assets/25473318/2f44bd57-f5de-4196-a1bb-35bf13ae4404">


I'll try to see if there are any quick fixes to handle those edge cases, but I'll close this as not planned for now, Thanks!

**Comment by tjbck** [comment]
Just updated the UI rendering logic, I can't guarantee that the UI will render LaTeX 100% of the time as per the reason mentioned above, but it should cover more edge cases and before, Thanks!

**Comment by YangSJ1** [comment]
Still it sometimes can't render latex properly, even when the expressions are in $$.

**Comment by jukofyork** [comment]
I've figured out this so far:

If the model outputs like this:

```
$$
<expression>
$$
```

or this:

```
\[
<expression>
\]
```

It won't display correctly.

But it can display:
```
$$ <expression> $$
```

It also seems able to use `\\( <expression \\)` for inline expressions  or:

```
\\[ <expression> \\]
```

BUT only if you double escape the `\` symbols!

I haven't found a way yet to get it to display `$ <expression> $` though.

I found this out by asking `GPT-4o` to try lots of different versions, but sadly however much I tried; I could find no system message that actually encourages models to use this format and they will nearly always just revert to the broken format until you remind them :/
